examples/handshake.rs

```diff
@@ -80,10 +80,7 @@ fn build_version_message(address: SocketAddr) -> message::NetworkMessage {
     let services = constants::ServiceFlags::NONE;
 
     // "standard UNIX timestamp in seconds"
-    let timestamp = SystemTime::now()
-        .duration_since(UNIX_EPOCH)
-        .expect("Time error")
-        .as_secs();
+    let timestamp = SystemTime::now().duration_since(UNIX_EPOCH).expect("Time error").as_secs();
```

Comment:
I prefer these to be always on a new line. The practical consequence is that you can quickly comment out things if types match and diffs are easier to understand. I had real-life experience with long iterator chain where types were preserved (lot of `map`s) and could easily experiment with the code by commenting/adding lines.

Reply:
IMO: If a chain of things is so long that it hinders your ability to see stuff, you have some other problems to address.

Like in this case the whole thing could be some `fn get_current_unix_ts_secs()`, because why would anyone wanted to parse this lower-level  4 element implementation detail, when looking at the higher level code.


It's an example code, so :shrug: , but this whole function could be more idiomatically:

```rust
fn build_version_message(address: SocketAddr) -> message::NetworkMessage {
    message::NetworkMessage::Version(
        VersionMessage::new()
          .service(ServiceFlags::NONE),
          .timestamp(get_current_unix_ts_secs() as i64),
          .recv_addr(Address::new(&address, ServiceFlags::NONE)),
          .from_addr(Address::new(&my_address, ServiceFlags::NONE)),
           /* ... */
          .build()
   )
}
```

with comments about what each thing means on the builder methoder, instead of each variable, and 1/4 of the LoC size, while being much easier to understand.

Reply:
BTW. I wish Rust just made a typed builder a language thing, because in the absence of optional and named arguments, builders are a common necessity.

Reply:
This is a positive discussion I can get into! The topic under discussion is what to set the value of `chain_width`, it is currently set to 80, the default is 60. The chosen value was in response to previous comments by @TheBlueMatt which he has restated here. Do any of the comments change your view @Kixunil or are you a hard vote for 'always on a new line'?

Reply:
Oh, awesome, another example of why units are important. Does `chain_width` represent the number of potatoes? :D

OK, I spent ~two minutes looking it up it, it's percentage of max width, which IMO is a weird unit. I'd say two dots are fine  on one line (`foo.bar().unwrap()`) but this can not be configured. :(

@dpc makes a good point that long chain may indicate a design problem. Actually, I think this applies to various other formatting issues - many arguments in a function, nested function calls... Apart from rustfmt I'd really like to have some guideline along "if rustfmt produces garbage look at redesigning the code" but it's hard (impossible?) to automatically detect/enforce.

I would go with the default of 60 for now, but again, if most people prefer 80, I will accept it.

Reply:
> it's percentage of max width

Bother, I remember reading this now you say it but yesterday I was thinking this was 'number of characters'. Since `max_width` is 100 this is effectively the same. I'm going to add a comment to the config file about it. 

And setting to `chain_width = 60`.  cc'ing @TheBlueMatt so he can jump in on this straight away if he feels like it :)

In the red corner, Martin 'short combinator chains for life' Habovštiak and in the blue corner Matt 'multi-lines piss me off' Corallo - ding ding ding.

Reply:
Meh, neither is great and if there even exists optimal line, I don't think it can be expressed in code. So I give up on this discussion, do as you wish.

Reply:
No worries @Kixunil, this whole rustfmt thing is a bit like that :).  This leaves use with @TheBlueMatt favours 100%, I probably lean towards 80% (not for any good reason except its half way between default and 100, seems like a compromise), default is 60%. Any one else want to chime in on this particular option?

Reply:
EDIT: Deleted my comment after seeing more discussion on this below.